modules/stochastic_tools/src/base/StochasticToolsApp.C

```diff
@@ -6,14 +6,22 @@
 #include "StateSimTester.h"
 #include "StateSimRunner.h"
 
-// distributions
+// Distributions
 #include "UniformDistribution.h"
-// samplers
+#ifdef LIBMESH_HAVE_BOOST
```

Comment:
I don't like this. The problem is that the error message you'll get when you don't have boost is that there's an unknown object name "WeibullDistrution" with the somewhat confusing error that we tried to load the Stochastic module and that object wasn't there. I think a better design would be to always have the object registered and have the API filled out to the minimum required for each of these BOOST-dependent objects. However you'll use the #ifdef define to just toggle a useful error message in the constructor when BOOST is missing. This will clean up the registration and make the error messages cleaner.